dvc/output/base.py

```diff
@@ -277,7 +277,8 @@ def save(self):
             self.verify_metric()
 
         if not self.use_cache:
-            self.hash_info = self.get_hash()
+            if self.repo:
+                self.hash_info = self.get_hash()
```

Comment:
From https://github.com/iterative/dvc/pull/5118#issuecomment-750709994 . Let's not intoroduce this change, as it is not necessary.
```suggestion
```